File: core/src/main/java/com/joshwiegand/abandon/managers/Jukebox.java
package com.joshwiegand.abandon.managers;

import java.util.*;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.audio.Sound;
import com.badlogic.gdx.files.FileHandle;

public class Jukebox {

  private static Map<String, Sound> SOUNDS = new HashMap<String, Sound>();
  private static final List<String> SOUND_NAMES = Arrays.asList("explode",
      "extralife",
      "largesaucer",
      "pulsehigh",
      "pulselow",
      "saucershoot",
      "shoot",
      "smallsaucer",
      "thruster");

  public static void load() {
    for (String soundName : SOUND_NAMES) {
      FileHandle soundFile = Gdx.files.internal("sounds/" + soundName + ".ogg");
      SOUNDS.put(soundName, Gdx.audio.newSound(soundFile));
    }
  }

  public static void play(String name) {
    SOUNDS.get(name).play();
  }

  public static void loop(String name) {
    SOUNDS.get(name).loop();
  }

  public static void stop(String name) {
    SOUNDS.get(name).stop();
  }

  public static void stopAll() {
    for (Sound s : SOUNDS.values()) {
      s.stop();
    }
  }

  public static void unload() {
    for (String key : SOUNDS.keySet()) {
      SOUNDS.get(key).dispose();
    }
  }
}

File: old/abandon/core/src/main/kotlin/com/joshwiegand/abandon/managers/Jukebox.kt
package com.joshwiegand.abandon.managers

import java.util.*

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.audio.Sound
import com.badlogic.gdx.files.FileHandle

object Jukebox {

    private val SOUNDS = HashMap<String, Sound>()
    private val SOUND_NAMES = Arrays.asList("explode",
            "extralife",
            "largesaucer",
            "pulsehigh",
            "pulselow",
            "saucershoot",
            "shoot",
            "smallsaucer",
            "thruster")

    fun load() {
        for (soundName in SOUND_NAMES) {
            val soundFile = Gdx.files.internal("sounds/$soundName.ogg")
            SOUNDS[soundName] = Gdx.audio.newSound(soundFile)
        }
    }

    fun play(name: String) {
        SOUNDS[name]?.play()
    }

    fun loop(name: String) {
        SOUNDS[name]?.loop()
    }

    fun stop(name: String) {
        SOUNDS[name]?.stop()
    }

    fun stopAll() {
        for (s in SOUNDS.values) {
            s.stop()
        }
    }

    fun unload() {
        for (key in SOUNDS.keys) {
            SOUNDS[key]?.dispose()
        }
    }
}